### PR TITLE
Wraps call in try/catch to avoid crash in dev

### LINKED ISSUE #464: Fix New Mobile Site Bug  - WETH to ETH
On Rinkeby, when I swap WETH to ETH I get the following bug:

image
image

### PR DESCRIPTION
**Pull request by robbestad**
Closes #464

### CODE REVIEW

File: src/hooks/useReferencePriceSubscriber.ts [hunk 1]
```diff
@@ -27,9 +27,15 @@ const useGasPriceSubscriber = () => {
...
-          dispatch(setFastGasPrice(price.toString()));
+        let price = "0";
+        try {
+          let _price = await getFastGasPrice();
+          if (_price) price = _price.toString();
+        } catch (e) {
+          console.error(e);
+        }
+        if (parseFloat(price)!==0) {
+          dispatch(setFastGasPrice(price));
```
Comment (dmosites) [inline src/hooks/useReferencePriceSubscriber.ts:38]: Figure this can be moved up to line 33 (replacing `price = _price.toString()`)

Comment (robbestad) [reply src/hooks/useReferencePriceSubscriber.ts:38]: Certainly. It looks much better that way

File: src/hooks/useReferencePriceSubscriber.ts [hunk 2]
```diff
@@ -27,9 +27,13 @@ const useGasPriceSubscriber = () => {
...
-        const price = await getFastGasPrice();
-        if (price) {
-          dispatch(setFastGasPrice(price.toString()));
+        try {
+          let price = await getFastGasPrice();
+          if (price) {
+            dispatch(setFastGasPrice(price.toString()));
+          }
+        } catch (e) {
+          console.error(e);
```
Comment (piersss) [inline src/hooks/useReferencePriceSubscriber.ts:36]: How about adding an extra function with comment to make it a little bit more readable?

```
// Fix for call not working in dev
const getFastGasPriceFix = async (): Promise<string> => {
  try {
    const price = await getFastGasPrice();
    return price ? price.toString() : "0";
  } catch (e) {
    console.error(e);
    return "0";
  }
}

-----

const updateGasPrice = async () => {
 const price = await getFastGasPriceFix();
 if (parseFloat(price)!==0) {
    dispatch(setFastGasPrice(price));
 }
};
```

Comment (piersss) [reply src/hooks/useReferencePriceSubscriber.ts:36]: Ok never mind!

**Review by piersss — APPROVED** [review]

**Pull request merged**